### PR TITLE
Extract module crypto

### PR DESCRIPTION
**Pull request by bmarty**

### CODE REVIEW

File: matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+package org.matrix.androidsdk.crypto.interfaces
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+import org.matrix.androidsdk.crypto.model.crypto.OlmPayloadContent
+import java.io.*
+import java.util.zip.GZIPOutputStream
+
+interface CryptoUtil {
```
Comment (ganfra) [inline matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt]: Is this really necessary? I think most of these methods should not come from outside but be hidden

File: matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoSession.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package org.matrix.androidsdk.crypto.interfaces
+
+interface CryptoSession {
+    val myUserId: String
+
+    fun getDeviceId(): String
+
+    fun setDeviceId(deviceId: String)
+
+    fun getCryptoRestClient(): CryptoCryptoRestClient
```
[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoSession.kt]: Crypto REST clients inherit from RestClient which is defined in the matrix-sdk module.

File: matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt]: was the fastest way to provided facilities from matrix-sdk to matrix-sdk-crypto

File: matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoSession.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoSession.kt]: ok, done

File: matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-crypto/src/main/java/org/matrix/androidsdk/crypto/interfaces/CryptoUtil.kt]: ok, I've removed this

**Comment by bmarty** [comment]
PR updated

**Pull request merged**